### PR TITLE
Use swap factory for for all halight dry tests

### LINKED ISSUE #2384: e2e tests: Use swap factory
## Problem

`swap_factory.ts` is use to create the body to create a swap on the new REST API routes.
However, some of the functions to default values for the body swap are present in `default.ts` and used by `actor.ts`.

Note that `swap_factory.ts` already support han-ethereum-ether-halight-lightning-bitcoin pair. Similar logic needs to be written for the other pairs.

## Goal

- Consolidate all functions to generate the body for the new cnd REST API routes in `swap_factory.ts`.
- Ensure all LN tests (e.g. dry tests and lnd sanity) use `SwapFactory`.
- This means the halight dry tests need to be converted to 2 actor tests.
- The expiries (and other info) to use for a swap are decided within the swap factory for all actors (and not each actor generate the expiry) as both actor need to inject the exact same in the create swap request to cnd REST API.

### PR DESCRIPTION
**Pull request by D4nte**
Resolves #2384.

### CODE REVIEW

File: api_tests/tests/ether_halight.ts [hunk 1]
```diff
@@ -10,13 +10,14 @@ import { sleep } from "../src/utils";
 it(
     "han-ethereum-ether-halight-lightning-bitcoin-alice-redeems-bob-redeems",
     twoActorTest(async ({ alice, bob }) => {
-        const [aliceBody, bobBody] = await SwapFactory.newSwap(alice, bob);
+        const bodies = (await SwapFactory.newSwap(alice, bob))
+            .hanEthereumEtherHalightLightningBitcoin;
```
Comment (thomaseizinger) [inline api_tests/tests/ether_halight.ts:14]: I've recently found that using a mix of `then` and await is a nice way of chaining things :)


```suggestion
        const bodies = await SwapFactory.newSwap(alice, bob).then(bodies => bodies.hanEthereumEtherHalightLightningBitcoin)
```

File: api_tests/src/actors/swap_factory.ts [hunk 2]
```diff
@@ -19,18 +19,36 @@ import {
...
-        bob: Actor
-    ): Promise<
-        [
-            HanEthereumEtherHalightLightningBitcoinRequestBody,
-            HanEthereumEtherHalightLightningBitcoinRequestBody
-        ]
-    > {
-        const ledgers: (keyof AllWallets)[] = ["ethereum", "lightning"];
+        bob: Actor,
+        dry?: boolean
```
Comment (thomaseizinger) [inline api_tests/src/actors/swap_factory.ts:30]: I am not seeing any use of the `dry` parameter?

File: api_tests/src/actors/swap_factory.ts [hunk 3]
```diff
@@ -19,18 +19,36 @@ import {
...
+    ): Promise<{
```
Comment (thomaseizinger) [inline api_tests/src/actors/swap_factory.ts:31]: Interesting idea to return all of them at once :)

File: api_tests/src/actors/swap_factory.ts (hunk 2 above)
Comment (D4nte) [reply api_tests/src/actors/swap_factory.ts:30]: in `lightning_routes.ts`

**Comment by thomaseizinger** [comment]
> wow... those type names sweat_smile

We'll find something better eventually :)

**Pull request merged**